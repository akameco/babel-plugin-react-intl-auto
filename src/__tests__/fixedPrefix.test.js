// @flow
import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from '..'

const filename = path.resolve(__dirname, '..', '__fixtures__', 'messages.js')

const defaultTest = {
  title: 'use static prefix instead of path to file',
  code: `
import { defineMessages } from 'react-intl'

export default defineMessages('customStaticPrefix', {
  hello: 'hello',
})
`,
}

const emptyStaticPrefixTest = {
  title: 'ignore empty static prefix',
  code: `
import { defineMessages } from 'react-intl'

export default defineMessages('', {
  hello: 'hello',
})
`,
}

const multiExportTest = {
  title: 'multi export',
  code: `
import { defineMessages } from 'react-intl'

export const extra = defineMessages('customStaticPrefix', {
  hello: 'hello world extra'
})

export default defineMessages('anotherCustomPrefix', {
  hello: 'hello world',
})
`,
}

const multiExportWithTheSamePrefixTest = {
  title: 'multi export with the same prefix',
  code: `
import { defineMessages } from 'react-intl'

export const extra = defineMessages('customStaticPrefix', {
  hello: 'hello world extra'
})

export default defineMessages('customStaticPrefix', {
  hello: 'hello world',
})
`,
}

const tests = [
  defaultTest,
  emptyStaticPrefixTest,
  multiExportTest,
  multiExportWithTheSamePrefixTest,
]

cases([{ title: 'default', tests }])

function cases(
  testCases: Array<{
    title: string,
    tests: $ReadOnlyArray<{ title: string, code: string }>,
  }>
) {
  const defaultOptions = {
    title: '',
    plugin,
    snapshot: true,
    babelOptions: { filename },
    tests: [],
  }
  for (const testCase of testCases) {
    testCase.tests = testCase.tests.map(t => ({ ...t, title: t.title }))
    pluginTester({ ...defaultOptions, ...testCase })
  }
}
