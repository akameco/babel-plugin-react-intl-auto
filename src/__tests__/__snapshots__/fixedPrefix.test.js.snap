// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default ignore empty static prefix: ignore empty static prefix 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages('', {
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`default multi export with the same prefix: multi export with the same prefix 1`] = `

import { defineMessages } from 'react-intl'

export const extra = defineMessages('customStaticPrefix', {
  hello: 'hello world extra'
})

export default defineMessages('customStaticPrefix', {
  hello: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export const extra = defineMessages({
  hello: {
    "id": "customStaticPrefix.hello",
    "defaultMessage": 'hello world extra'
  }
});
export default defineMessages({
  hello: {
    "id": "customStaticPrefix.hello",
    "defaultMessage": 'hello world'
  }
});

`;

exports[`default multi export: multi export 1`] = `

import { defineMessages } from 'react-intl'

export const extra = defineMessages('customStaticPrefix', {
  hello: 'hello world extra'
})

export default defineMessages('anotherCustomPrefix', {
  hello: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export const extra = defineMessages({
  hello: {
    "id": "customStaticPrefix.hello",
    "defaultMessage": 'hello world extra'
  }
});
export default defineMessages({
  hello: {
    "id": "anotherCustomPrefix.hello",
    "defaultMessage": 'hello world'
  }
});

`;

exports[`default use static prefix instead of path to file: use static prefix instead of path to file 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages('customStaticPrefix', {
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "customStaticPrefix.hello",
    "defaultMessage": 'hello'
  }
});

`;
